test/utils.py

```diff
@@ -0,0 +1,51 @@
+from collections.abc import Sequence
+from dataclasses import dataclass
+
+import numpy as np
+from gymnasium import spaces
+
+from tianshou.data.collector import CollectStats
+
+
+@dataclass
+class SpaceInfo:
+    action_shape: int | Sequence[int]
+    state_shape: int | Sequence[int]
+    action_dim: int
+    state_dim: int
+    min_action: float
+    max_action: float
+
+
+def get_space_info(
+    space: spaces.Space,
+) -> SpaceInfo:
+    if isinstance(space, spaces.Box):
+        return SpaceInfo(
+            action_shape=space.shape,
+            state_shape=space.shape,
+            action_dim=space.shape[0],
+            state_dim=space.shape[0],
+            min_action=float(np.min(space.low)),
+            max_action=float(np.max(space.high)),
+        )
+    elif isinstance(space, spaces.Discrete):
+        return SpaceInfo(
+            action_shape=int(space.n),
+            state_shape=int(space.n),
+            action_dim=int(space.n),
+            state_dim=int(space.n),
+            min_action=float(space.start),
+            max_action=float(space.start + space.n - 1),
+        )
+    else:
+        raise NotImplementedError("Unsupported space type")
+
+
+def print_final_stats(collect_stats: CollectStats) -> None:
```

Comment:
where is this used? print statements in tests are not really useful, I think, and it's not importable in the rest of the code

Reply:
Yes, the CollectStats should inherit from the ToStringMixin or a dedicated Mixin that adds a suitable `__repr__`. I'd do that in a followup PR. For now, printing info like this in tests should be discouraged. If you want to fix it, please use a dedicated `log = logging.getLogger(__name__)` in each module and then `log.info` or `log.debug` instead of print. Also, you can just use the CollectStats repr, assuming that it will be improved soon. No need for a separate printing util. Feel free to open an issue for a better repr for CollectStats (and other things) if you want :)

Reply:
> Also, you can just use the CollectStats repr, assuming that it will be improved soon. No need for a separate printing util.

Yep, I'll do this and open a separate PR afterwards with the improvement.

Reply:
I kinda forgot that the solution was on his branch and not on master yet :sweat_smile: